lib/storage/metadata/mongoclient/MongoClientInterface.ts

```diff
@@ -623,7 +633,13 @@ class MongoClientInterface {
             },
         }, {
             upsert: true,
-        }).then(() => cb(null)).catch(err => {
+        }).then(result => {
+            if (result.modifiedCount === 0 && result.upsertedCount === 0) {
```

Comment:
we can have a case where we try to update with the same value , then we can have the operation succeeding but the upserted, and modified counts to 0 but it's still not an internal error ?  ( the case where the bucket is created with capabilities already... )

Reply:
we can add a check for the acknowledged in this case 
```suggestion
            if (result.acknowledged && result.modifiedCount === 0 && result.upsertedCount === 0) {
```